src/components/media-player/media-player.service.ts

```diff
@@ -138,7 +138,7 @@ export class MediaPlayerService {
     }
 
     get duration(): number {
-        return this._mediaPlayer ? this._mediaPlayer.duration : 0;
+        return this._mediaPlayer && !isNaN(this.mediaPlayer.duration) ? this._mediaPlayer.duration : 0;
```

Comment:
Is this change related to this ticket?

Reply:
There was an expression changed error happening, this was to prevent it I believe

Reply:
Yeah there was a ExpressionChangedAfterItHasBeenChecked error when running fixture.detectChanges() in the karma tests.